nhais/outbound/request/tests/test_acceptance_amendment.py

```diff
@@ -0,0 +1,132 @@
+import tornado.testing
+from unittest.mock import patch, MagicMock
+
+from tornado.web import Application
+
+from outbound.request.acceptance_amendment import AcceptanceAmendmentRequestHandler
+
+import json
+import os
+
+from mesh.mesh_outbound import MeshOutboundWrapper
+
+root_dir = os.path.dirname(os.path.abspath(__file__))
+
+REQUEST_PATIENT_URI = "/fhir/Patient/123"
+REQUEST_VALID_OPERATION = "Patient"
+REQUEST_INVALID_OPERATION = "Patientssss"
+
+valid_json_patient = root_dir + "/../../../common/data/patient.json"
+with open(valid_json_patient) as file:
+    data = json.load(file)
+VALID_REQUEST_BODY = json.dumps(data)
+
+invalid_json_patient_id = root_dir + "/../../../common/data/patient_invalid_id.json"
+with open(invalid_json_patient_id) as file:
+    data = json.load(file)
+INVALID_ID_REQUEST_BODY = json.dumps(data)
+
+missing_json_patient_id = root_dir + "/../../../common/data/patient_missing_id.json"
+with open(missing_json_patient_id) as file:
+    data = json.load(file)
+MISSING_ID_REQUEST_BODY = json.dumps(None)
+
+
+class TestAcceptanceAmendmentRequestHandler(tornado.testing.AsyncHTTPTestCase):
+
+    def get_app(self) -> Application:
+        return tornado.web.Application([(r'/fhir/Patient/(.*)', AcceptanceAmendmentRequestHandler)])
+
+    async def async_magic(self):
+        pass
+
+    def test_invalid_post_request_line_return_404_response_code(self):
+        response = self.fetch(r'/water/Panda/9000000009', method="POST",
+                              body=VALID_REQUEST_BODY)
+        self.assertEqual(404, response.code)
+        self.assertEqual('Not Found', response.reason)
+
+    @patch.object(MeshOutboundWrapper, "send")
+    @patch.object(MeshOutboundWrapper, "__init__")
+    def test_happy_path(self, mock_init, mock_send):
+        mock_init.return_value = None
+        MagicMock.__await__ = lambda x: self.async_magic().__await__()
+
+        response = self.fetch(r'/fhir/Patient/9000000009', method="POST",
+                              body=VALID_REQUEST_BODY)
+        self.assertEqual(202, response.code)
+
+    @patch.object(MeshOutboundWrapper, "send")
+    @patch.object(MeshOutboundWrapper, "__init__")
+    def test_invalid_payload_id_not_string(self, mock_init, mock_send):
+        mock_init.return_value = None
+        MagicMock.__await__ = lambda x: self.async_magic().__await__()
+
+        response = self.fetch('/fhir/Patient/9000000009', method="POST",
+                              body=INVALID_ID_REQUEST_BODY)
+
+        response_body = json.loads(response.body.decode())
```

Comment:
Would it be better to create an fhir.resources OperationOutcome from the response and validate property values from that?

Reply:
All tests now create operation outcome object